Sources/Apollo/ApolloClient.swift

```diff
@@ -123,6 +123,21 @@ public class ApolloClient {
     return send(operation: mutation, context: context, handlerQueue: queue, resultHandler: resultHandler)
   }
 
+  /// Performs a mutation by sending it to the server.
+  ///
+  /// - Parameters:
+  ///   - mutation: The mutation to perform.
+  ///   - queue: A dispatch queue on which the result handler will be called. Defaults to the main queue.
+  ///   - resultHandler: An optional closure that is called when mutation results are available or when an error occurs.
+  /// - Returns: An object that can be used to cancel an in progress mutation.
+  @discardableResult public func perform<Mutation: GraphQLUploadMutation>(mutation: Mutation, queue: DispatchQueue = DispatchQueue.main, resultHandler: OperationResultHandler<Mutation>? = nil) -> Cancellable {
```

Comment:
I think we should be able to avoid duplicating `perform` and related methods by removing `GraphQLUploadMutation` and relying on protocol composition instead.